### PR TITLE
Fix storyboard not showing when user dim is disabled

### PR DESCRIPTION
**Pull request by frenzibyte**
Forgotten check if user dim is enabled or not

### CODE REVIEW

**Comment by peppy** [comment]
Where is this visible/testable?

**Comment by iiSaLMaN** [comment]
you mean where it happens? i've encountered that while working on PR #5639

**Comment by peppy** [comment]
Encountered where? How? I'm trying to validate this change but have no idea where this actually has meaning.

**Comment by iiSaLMaN** [comment]
> Encountered where? How? I'm trying to validate this change but have no idea where this actually has meaning.

When you play a beatmap with storyboard and disable user dim, if the dim level is less than 1, the content will be hidden, this PR adds a check if user dim is enabled or not

**Comment by peppy** [comment]
How do you "disable user dim"?

**Comment by iiSaLMaN** [comment]
> How do you "disable user dim"?

`EnableUserDim.Value = false`?

**Comment by peppy** [comment]
so i need to add code triggered by something during gameplay to see this change?

**Comment by iiSaLMaN** [comment]
> so i need to add code triggered by something during gameplay to see this change?

you can just disable it by changing default value of `UserDimContainer.EnableUserDim` and set user dim level to 0 or make `ContentDisplayed` false, you'll see the change

**Comment by peppy** [comment]
So the correct answer to my first question is "this is not seen anywhere and currently untestable", right?

**Comment by iiSaLMaN** [comment]
yep, will add a test a bit later, the purpose of this PR is to fix a bug i found which affects #5639

**Comment by iiSaLMaN** [comment]
@peppy Tests added, `DisableUserDimStoryboardTest` in `TestSceneUserDimContainer`, added checks if content is displayed or not, you can see the change now.

**Comment by nyquillerium** [comment]
I'm concerned that this change is fixing a problem in `DimmableStoryboard` but you're applying a change to the base `UserDimContainer` class.

**Comment by iiSaLMaN** [comment]
> I'm concerned that this change is fixing a problem in `DimmableStoryboard` but you're applying a change to the base `UserDimContainer` class.

It would be a problem to any dimmable that would hide their content if user dim is 1

File: osu.Game/Graphics/Containers/UserDimContainer.cs [hunk 1]
```diff
@@ -76,9 +76,9 @@ protected override void LoadComplete()
         /// </summary>
         protected virtual void UpdateVisuals()
         {
-            ContentDisplayed = ShowDimContent;
+            ContentDisplayed = !EnableUserDim.Value || ShowDimContent;
```
Comment (nyquillerium) [inline osu.Game/Graphics/Containers/UserDimContainer.cs]: You can probably move this to the default of `ShowDimContent`

Comment (nyquillerium) [reply osu.Game/Graphics/Containers/UserDimContainer.cs]: I think you forgot to delete this

Comment (iiSaLMaN) [reply osu.Game/Graphics/Containers/UserDimContainer.cs]: Whoops

**Comment by iiSaLMaN** [comment]
After a bit of rethinking, I found a better solution that'll go well with a later PR (adding `DimAmount`), which is adding a property `DimLevel` that calculates the actual dim level with user dim enabled or disabled, and `DimmableStoryboard.ShowDimContent` will depend on that.

File: osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs [hunk 2]
```diff
@@ -149,10 +149,10 @@ public void StoryboardTransitionTest()
         }
 
         /// <summary>
-        /// Check if the <see cref="UserDimContainer"/> is properly accepting user-defined visual changes at all.
+        /// Check if the <see cref="UserDimContainer"/> is properly accepting user-defined visual changes in background at all.
```
Comment (peppy) [inline osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs]: i don't get this change

File: osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs [hunk 3]
```diff
@@ -165,6 +165,28 @@ public void DisableUserDimTest()
...
 
+        /// <summary>
+        /// Check if the <see cref="UserDimContainer"/> is properly accepting user-defined visual changes in storyboard at all.
+        /// </summary>
+        [Test]
+        public void DisableUserDimStoryboardTest()
+        {
+            performFullSetup();
+            createFakeStoryboard();
+            AddStep("Storyboard Enabled", () =>
```
Comment (peppy) [inline osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs]:
```suggestion
            AddStep("enable storyboard", () =>
```

File: osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs [hunk 4]
```diff
@@ -165,6 +165,28 @@ public void DisableUserDimTest()
             AddAssert("Screen is dimmed and blur applied", () => songSelect.IsBackgroundDimmed() && songSelect.IsUserBlurApplied());
         }
 
+        /// <summary>
+        /// Check if the <see cref="UserDimContainer"/> is properly accepting user-defined visual changes in storyboard at all.
```
Comment (peppy) [inline osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs]:
```suggestion
        /// Ensure <see cref="UserDimContainer"/> is properly accepting user-defined visual changes for a storyboard.
```

File: osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs (hunk 2 above)
Comment (peppy) [inline osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs]:
```suggestion
        /// Ensure <see cref="UserDimContainer"/> is properly accepting user-defined visual changes for a background.
```

File: osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs [hunk 5]
```diff
@@ -165,6 +165,28 @@ public void DisableUserDimTest()
...
+        public void DisableUserDimStoryboardTest()
+        {
+            performFullSetup();
+            createFakeStoryboard();
+            AddStep("Storyboard Enabled", () =>
+            {
+                player.ReplacesBackground.Value = true;
+                player.StoryboardEnabled.Value = true;
+            });
+            AddStep("EnableUserDim enabled", () => player.DimmableStoryboard.EnableUserDim.Value = true);
```
Comment (peppy) [inline osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs]: enable user dim

File: osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs [hunk 6]
```diff
@@ -165,6 +165,28 @@ public void DisableUserDimTest()
...
+            AddStep("Set dim level to 1", () => songSelect.DimLevel.Value = 1f);
+            waitForDim();
+            AddAssert("Storyboard is invisible", () => !player.IsStoryboardVisible);
+            AddStep("EnableUserDim disabled", () => player.DimmableStoryboard.EnableUserDim.Value = false);
```
Comment (peppy) [inline osu.Game.Tests/Visual/Background/TestSceneUserDimContainer.cs]: disable user dim

**Review by peppy — CHANGES_REQUESTED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**